arcgis-ios-sdk-samples/Maps/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningViewController.swift

```diff
@@ -0,0 +1,306 @@
+        let locationDataSource = AGSIndoorsLocationDataSource(positioningTable: positioningTable, pathwaysTable: pathwaysTable, positioningID: globalID)
+        // The delegate which will receive location and status updates
+        // from the data source.
+        locationDataSource.locationChangeHandlerDelegate = self
+        self.indoorsLocationDataSource = locationDataSource
+        if let extent = pathwaysTable.extent {
+            mapView.setViewpointGeometry(extent)
+        }
+        mapView.locationDisplay.dataSource = locationDataSource
+        mapView.locationDisplay.autoPanMode = .navigation
+        // Asynchronously start of the location display, which will in-turn
+        // start `indoorsLocationDataSource` to receive IPS updates.
+        mapView.locationDisplay.start { [weak self] (error) in
+            guard let self = self, let error = error else { return }
+            self.presentAlert(error: error)
+        }
+    }
+    
+    /// A helper method to load one feature table at a time. Stop loading if
+    /// error occurs.
+    /// - Parameters:
+    ///   - tables: The feature tables to load.
+    ///   - completion: The load result for a table.
+    func loadTables<C>(_ tables: C, completion: @escaping (Result<Void, Error>) -> Void) where C: RandomAccessCollection, C.Element == AGSFeatureTable {
+        guard let table = tables.last else {
+            completion(.success(()))
+            return
+        }
+        table.load { [weak self] error in
+            if let error = error {
+                completion(.failure(error))
+            } else {
+                self?.loadTables(tables.dropLast(), completion: completion)
+            }
+        }
+    }
+    
+    /// Display features on a certain floor level using definition expression.
+    /// - Parameter floor: The floor level of the features to be displayed.
+    func displayFeatures(onFloor floor: Int) {
+        (mapView.map!.operationalLayers as! [AGSLayer]).forEach { layer in
+            if layer.name == "Details" || layer.name == "Units" || layer.name == "Levels" {
+                if let featureLayer = layer as? AGSFeatureLayer {
+                    featureLayer.definitionExpression = "VERTICAL_ORDER = \(floor)"
+                }
+            }
+        }
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["ShowDeviceLocationUsingIndoorPositioningViewController"]
+    }
+    
+    deinit {
+        // Stop location display, which in turn stop the data source.
+        mapView.locationDisplay.stop()
+        // Reset indoors location data source.
+        indoorsLocationDataSource?.locationChangeHandlerDelegate = nil
+        indoorsLocationDataSource = nil
```

Comment:
Neither of these is necessary. `locationChangeHandlerDelegate` is a`weak` property, so it will automatically be set to `nil` when the delegate (`self`) goes away. And setting `indoorsLocationDataSource` to `nil` is pointless because that property won't ever be accessed again as `self` is in the process of going away. In other words, this will happen automatically as part of deinitialization.

```suggestion
```